src/plot.js

```diff
@@ -99,6 +101,18 @@ export function plot(options = {}) {
     const {data, facets, channels} = mark.initialize(markFacets, facetChannels);
     applyScaleTransforms(channels, options);
     stateByMark.set(mark, {data, facets, channels});
+
+    // Warn for the common pitfall of wanting to facet mapped data.
+    if (
+      facetIndex?.length > 1 && // non-trivial faceting
+      mark.facet === "auto" && // no explicit mark facet option
+      mark.data !== facet.data && // mark not implicitly faceted (different data)
+      arrayify(mark.data)?.length === facetDataLength // mark data seems parallel to facet data
+    ) {
+      warn(
+        `Warning: did you want to facet the ${mark.ariaLabel} mark? The mark data and facet data are distinct, but they have the same length. If this mark should be faceted, set the mark facet option to true; otherwise, suppress this warning by setting the mark facet option to false.`
```

Comment:
Can we revert to my wording please? I don’t like the use of a question, and this no longer mentions that facet "auto" is in use.

Reply:
Happy to, I had just merged in @Fil's PR on top of your suggestion, happy to revert.